diff --git a/includes/view.inc b/includes/view.inc
index d082a0b..faf9f84 100644
--- a/includes/view.inc
+++ b/includes/view.inc
@@ -1150,7 +1150,7 @@ class view extends views_db_object {
       $cache = $this->display_handler->get_plugin('cache');
     }
     if ($cache && $cache->cache_get('results')) {
-      if($this->query->pager->use_pager()) {
+      if($this->query->pager->use_pager() || !empty($this->get_total_rows)) {
         $this->query->pager->total_items = $this->total_rows;
         $this->query->pager->update_page_info();
       }